lib/services/utils/ProcessReporter.ts

```diff
@@ -0,0 +1,99 @@
+import {IRailWayStation} from '../../interfaces/core';
+import {IReportRecord} from '../../interfaces';
+
+export class ProcessReporter {
+    private handlers: Map<string, any> = new Map();
+
+    /**
+     * @param {string} path
+     * @param {IRailWayStation} station
+     */
+    public registerHandler(path: string, station: IRailWayStation) {
+        this.handlers.set(
+            this.generateProcessId(station),
+            {
+                path,
+                station,
+            }
+        );
+    }
+
+    /**
+     * @param {IRailWayStation} station
+     * @param {IReportRecord | undefined} report
+     * @param options
+     */
+    public setReport(
+        station: IRailWayStation,
+        report: IReportRecord[] | void,
+        options: any,
+    ) {
+        const handler = this.handlers.get(this.generateProcessId(station));
+
+        if (handler) {
+            handler.report = report;
+            handler.options = options;
+        }
+    }
+
+    /**
+     * @return {any}
```

Comment:
Add comment and define type of the result. Don't use `any` if you 100% sure with what you're working